[TagInput] leftIconName + rightElement

giladgray opened the pull request:
#### Addresses #1412 

#### Changes proposed in this pull request:

- `leftIconName` renders an icon as first child
- `rightElement` appears as last child
- just like `InputGroup`

<img width="414" alt="screen shot 2017-08-07 at 8 47 27 pm" src="https://user-images.githubusercontent.com/464822/29055488-a910684a-7bb1-11e7-8d00-d4fc1f75d116.png">

cmslewis commented:
@giladgray can we also add support for `pt-round` in this PR, while we're on the topic of `TagInput` aesthetic extensions? That would have been the final necessary style piece for our side project last week.

cmslewis commented:
FYI Clear button doesn't look centered in the button in your screenshot: 
![image](https://user-images.githubusercontent.com/443450/29091912-45f5c86c-7c39-11e7-844a-0aca1f6c4631.png)

cmslewis commented on packages/labs/src/components/tag-input/tagInput.tsx:
```diff
@@ -149,10 +159,24 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
+        }
+        const iconClass = classNames(
+            Classes.TAG_INPUT_ICON,
+            useLarge ? CoreClasses.ICON_LARGE : CoreClasses.ICON_STANDARD ,
```
nit: no space before comma

giladgray commented:
addressed all the issues: button centering and padding and whitespace.

@cmslewis i poked around `.pt-round` support and it's not so easy. i think we should instead try to merge the `.pt-input-group` and `.pt-tag-input` styles and ensure that they both support the same things, _forever_.

llorca commented on packages/labs/src/components/tag-input/_tag-input.scss:
```diff
@@ -9,39 +9,56 @@
 
-$tag-input-padding: ($pt-input-height - $tag-height) / 2;
+$ti-padding: ($pt-input-height - $tag-height) / 2;
+$ti-right-padding: ($pt-input-height - $pt-button-height-small) / 2;
```
So much nicer with this added 👏 Perfectly squared off. `$tag-input-` ain't that long is it?

How about `-right-padding` -> `-padding-right` to match the property name?

giladgray commented:
👀 on https://github.com/palantir/blueprint/pull/1420/commits/e042aa84c4f73fa7242aafcd2e95278d40c70621 would be appreciated, to make sure nothing is weird
